src/applications/search/containers/SearchApp.jsx

```diff
@@ -383,22 +407,24 @@ class SearchApp extends React.Component {
       </div>
     );
 
-    if (hasErrors && !loading) {
+    if ((hasErrors && !loading) || userInput.length >= 255) {
+      const errorMessage =
+        userInput.length >= 255
+          ? 'The search is over the character limit. Shorten the search and try again.'
+          : `We’re sorry. Something went wrong on our end, and your search
+      didn't go through. Please try again`;
+
       return (
-        <div className="columns error">
+        <div className="columns error vads-u-margin-bottom--4">
           {/* this is the alert box for when searches fail due to server issues */}
           <va-alert status="error" data-e2e-id="alert-box">
-            <h3 slot="headline">Your search didn't go through</h3>
-            <div>
-              We’re sorry. Something went wrong on our end, and your search
-              didn't go through. Please try again
-            </div>
+            <h2 slot="headline">Your search didn't go through</h2>
+            <p>{errorMessage}</p>
           </va-alert>
           {searchInput}
         </div>
       );
     }
-
```

Comment:
Nitpick **Consider:** we should leave blank lines before `return`s to increase readability.